protocol/x/clob/keeper/deleveraging_test.go

```diff
@@ -1095,6 +1096,211 @@ func TestProcessDeleveraging(t *testing.T) {
 				*tc.offsettingSubaccount.GetId(),
 				uint32(0),
 				tc.deltaQuantums,
+				false,
+			)
+			if tc.expectedErr == nil {
+				require.NoError(t, err)
+
+				actualLiquidated := ks.SubaccountsKeeper.GetSubaccount(ks.Ctx, *tc.liquidatedSubaccount.GetId())
+				require.Equal(
+					t,
+					tc.expectedLiquidatedSubaccount,
+					actualLiquidated,
+				)
+
+				actualOffsetting := ks.SubaccountsKeeper.GetSubaccount(ks.Ctx, *tc.offsettingSubaccount.GetId())
+				require.Equal(
+					t,
+					tc.expectedOffsettingSubaccount,
+					actualOffsetting,
+				)
+			} else {
+				require.ErrorContains(t, err, tc.expectedErr.Error())
+			}
+		})
+	}
+}
+
+func TestProcessDeleveragingAtOraclePrice(t *testing.T) {
+	tests := map[string]struct {
+		// Setup.
+		liquidatedSubaccount satypes.Subaccount
```

Comment:
nit: let's use a different name since we are not really liquidating this subaccount.

Reply:
sure - this is test only so will let you make the call here